src/iControlLBO/iControlLBO.psm1

```diff
@@ -49,20 +60,38 @@ function New-LBConnection
 		[Parameter(Mandatory=$true)]
 		[string]$username,
 
-		[string]$password
+		[string]$password,
+
+		[string]$key
 	)
 
-	if ($password -eq $null)
+	if ([string]::IsNullOrEmpty($password))
 	{
-		$serverCredential = Get-Credential -Username $username
+		if ($key -eq $null)
```

Comment:
The if test for the key determines whether the the given password is used with the username, or as the password for the private key. It could probably be reworked, but I think it's valuable to distinguish in a prompt what the user is entering their password for, see line 72 and 76 below.

That said I'm open to changes.